### PR TITLE
Add documentation for kuromoji_completion filter

### PR DESCRIPTION
**Pull request by msfroh**
### Description
Adds a documentation line for the `kuromoji_completion` filter, and updates links to Lucene docs to `9_10_0` paths.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6552

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by msfroh** [comment]
Huh... Vale doesn't like my use of the word "romanized", but it's a perfectly cromulent word: https://en.wikipedia.org/wiki/Romanization

**Comment by hdhalter** [comment]
Thank you, @msfroh !

File: _analyzers/token-filters/index.md [hunk 1]
```diff
@@ -13,52 +13,53 @@ Token filters receive the stream of tokens from the tokenizer and add, remove, o
...
+`keyword_repeat` | [KeywordRepeatFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/KeywordRepeatFilter.html) | Emits each incoming token twice: once as a keyword and once as a non-keyword.
+`kstem` | [KStemFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/en/KStemFilter.html) | Provides kstem-based stemming for the English language. Combines algorithmic stemming with a built-in dictionary.
+`kuromoji_completion` | [JapaneseCompletionFilter](https://lucene.apache.org/core/9_10_0/analysis/kuromoji/org/apache/lucene/analysis/ja/JapaneseCompletionFilter.html) | Adds Japanese romanized terms to the token stream (in addition to the original tokens). Usually used to support autocomplete on Japanese search terms. Note that the filter has a `mode` parameter, which should be `index` when used in an index analyzer and `query` when used in a search analyzer. Requires the `analysis-kuromoji` plugin.
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/index.md:38]:
```suggestion
`kuromoji_completion` | [JapaneseCompletionFilter](https://lucene.apache.org/core/9_10_0/analysis/kuromoji/org/apache/lucene/analysis/ja/JapaneseCompletionFilter.html) | Adds Japanese romanized terms to the token stream (in addition to the original tokens). Usually used to support autocomplete on Japanese search terms. Note that the filter has a `mode` parameter, which should be set to `index` when used in an index analyzer and `query` when used in a search analyzer. Requires the `analysis-kuromoji` plugin. For information about installing the plugin, see [Additional plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/#additional-plugins).
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @msfroh! Just one suggestion.

File: _analyzers/token-filters/index.md (hunk 1 above)
Comment (natebower) [reply _analyzers/token-filters/index.md:38]: LGTM

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**